ThunderCloud/StormLink.swift

```diff
@@ -12,239 +12,266 @@ import Foundation
-	public var appIdentity: AppIdentity? {
-		guard let appIdentityIdentifier = appIdentityIdentifier else { return nil }
-		return AppLinkController().apps.first(where: {$0.identifier == appIdentityIdentifier })
-	}
-	
-	/// The URL to be passed to the recieving app
-	///
-	/// Only valid for inter-app links
-	public let destination: String?
-	
-	//MARK: - Timer links -
-	
-	/// The number of seconds the timer should run for
-	///
-	/// Only valid for timer links
-	public let duration: TimeInterval?
-	
-	//MARK: - Miscellaneous -
-	
-	/// Aribtrary attributes added to the link
-	public var attributes: [String] = []
+    
+    public enum LinkClass: String {
+        case app = "AppLink"
+        case call = "CallLink"
+        case email = "EmailLink"
+        case emergency = "EmergencyLink"
+        case external = "ExternalLink"
+        case `internal` = "InternalLink"
+        case localised = "LocalisedLink"
+        case native = "NativeLink"
+        case sms = "SmsLink"
+        case share = "ShareLink"
+        case timer = "TimerLink"
+        case unknown
+        case uri = "UriLink"
+        case url
+    }
+    
+    //MARK: - Init methods -
+    
+    /// Initialises with a CMS representation of a link
+    ///
+    /// - Parameter dictionary: A Link dictionary that you intend to push the user to
+    public required init?(dictionary: [AnyHashable : Any]) {
+        // Keep this around otherwise get compiler errors using `linkClass` before all members initialised
+        let _linkClass = LinkClass(rawValue: dictionary["class"] as? String ?? "unknown") ?? .unknown
+        linkClass = _linkClass
+        
+        super.init()
+        
+        configure(with: dictionary, languageController: StormLanguageController.shared)
+        
+        guard url != nil
+            || linkClass == .sms
+            || linkClass == .emergency
+            || linkClass == .share
+            || linkClass == .timer
+            || linkClass == .external
+            || linkClass == .uri else {
+                return nil
+        }
+    }
+    
+    public init?(dictionary: [AnyHashable: Any], languageController: StormLanguageController = StormLanguageController.shared) {
+        // Keep this around otherwise get compiler errors using `linkClass` before all members initialised
+        let _linkClass = LinkClass(rawValue: dictionary["class"] as? String ?? "unknown") ?? .unknown
+        linkClass = _linkClass
+        
+        super.init()
+        
+        configure(with: dictionary, languageController: languageController)
+        
+        guard url != nil
+            || linkClass == .sms
+            || linkClass == .emergency
+            || linkClass == .share
+            || linkClass == .timer
+            || linkClass == .external
+            || linkClass == .uri else {
+                return nil
+        }
+    }
+    
+    public func configure(with dictionary: [AnyHashable: Any], languageController: StormLanguageController) {
```

Comment:
@simonmitchell the changes in here (and this file) allow for the `languageController` instance to be injected so the title can be localised. This hasn't caused any of the existing tests to fail :)